data_model/derive/src/lib.rs

```diff
@@ -9,6 +10,49 @@ use iroha_macro_utils::Emitter;
 use manyhow::{emit, manyhow, Result};
 use proc_macro2::TokenStream;
 
+/// Construct a matching enum with references in place of enum variant fields
+///
+/// # Example
+///
+/// ```
+/// use iroha_data_model_derive::EnumRef;
+/// use parity_scale_codec::Encode;
+///
+/// #[derive(EnumRef)]
+/// #[enum_ref(derive(Encode))]
+/// pub enum InnerEnum {
+///     A(u32),
+///     B(i32)
+/// }
+///
+/// #[derive(EnumRef)]
+/// #[enum_ref(derive(Encode))]
+/// pub enum OuterEnum {
+///     A(String),
+///     #[enum_ref(transparent)]
+///     B(InnerEnum),
+/// }
+///
+/// /* will produce:
+/// pub(crate) enum InnerEnumRef<'a> {
+///     A(&'a u32),
+///     B(&'a i32),
+/// }
+///
+/// pub(crate) enum OuterEnumRef<'a> {
+///     A(&'a String),
+///     B(InnerEnumRef<'a>),
+/// }
+/// */
```

Comment:
```suggestion
/// /* will produce:
/// #[derive(Encode)
/// pub(crate) enum InnerEnumRef<'a> {
///     A(&'a u32),
///     B(&'a i32),
/// }
///
/// #[derive(Encode)
/// pub(crate) enum OuterEnumRef<'a> {
///     A(&'a String),
///     B(InnerEnumRef<'a>),
/// }
/// */
```

I think it should be with derives